src/main/java/com/mbcoder/iot/weatherstation/Controller.java

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright 2023 Esri
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.mbcoder.iot.weatherstation;
+
+import com.esri.arcgisruntime.data.Feature;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.pi4j.Pi4J;
+import com.pi4j.context.Context;
+
+import eu.hansolo.medusa.FGauge;
+import eu.hansolo.medusa.Gauge;
+import eu.hansolo.medusa.GaugeBuilder;
+import eu.hansolo.medusa.GaugeDesign;
+import eu.hansolo.medusa.SectionBuilder;
+
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import java.net.URL;
+import java.util.Random;
+import java.util.ResourceBundle;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import one.microproject.rpi.hardware.gpio.sensors.BME280;
+import one.microproject.rpi.hardware.gpio.sensors.BME280Builder;
+
+public class Controller implements Initializable {
+
+  @FXML VBox vBox;
+  private final UUID weatherStationID = UUID.fromString("db12edca-7d28-45e6-84a3-9484c6e50d12"); // GlobalID of weather station for this unit
+  @FXML private Gauge humidityGauge;
+  @FXML private Gauge digitalTempGauge;
+  @FXML private Gauge barometerGauge;
+  private Timer recordingTimer; // timer for reading from the sensors
+  private Timer loggingTimer; // timer for logging data to the feature service
+  private final boolean simulatedMode = false;
+  private double currentTemperature = 0;
+  private double currentPressureMb = 0;
+  private double currentHumidity = 0;
+  private BME280 sensor;
+  private ServiceFeatureTable reportTable;
+  private Context context;
+
+  /**
+   * Initializes the application on start up
+   * <p>
+   * @param url
+   * The location used to resolve relative paths for the root object, or
+   * {@code null} if the location is not known.
+   *
+   * @param resourceBundle
+   * The resources used to localize the root object, or {@code null} if
+   * the root object was not localized.
+   */
+  @Override
+  public void initialize(URL url, ResourceBundle resourceBundle) {
+
+    // set up the UI
```

Comment:
since the UI is being set up in FXML initially, just making this comment more accurate for the method which is setting up the barometer gauge

```suggestion
    // set up the UI for the barometer gauge
```